lib/metasploit/framework/login_scanner/zabbix.rb

```diff
@@ -115,14 +115,15 @@ def try_login(credential)
           res = try_credential(credential)
           if res && res.code == 302
             opts = {
-              'uri'     => normalize_uri('profile.php'),
+              #'uri'     => normalize_uri('profile.php'), --> works for versions 3 and 4 but not 5
+              'uri'     => normalize_uri('discoveryconf.php'),
               'method'  => 'GET',
               'headers' => {
                 'Cookie'  => "zbx_sessionid=#{self.zsession}"
               }
             }
             res = send_request(opts)
-            if (res && res.code == 200 && res.body.to_s =~ /<title>Zabbix .*: User profile<\/title>/)
+            if (res && res.code == 200 && res.body.to_s =~ /Zabbix/)
```

Comment:
See my comment above, I can easily bypass this Regex and trigger a false positive by having the word `Zabbix` anywhere within the body of my server's response. This regex needs to be a lot tighter, especially as it its in library code that could be reused by other modules.

Reply:
See my comment above.  I did test using the regex searching for the Author string and it tested successfully 
`            if res.body.to_s !~ /name="Author" content="Zabbix SIA"/`

Not sure if this matters, but according to the Zabbix repository page, versions 5.X are the only ones listed as supported.

Reply:
See my comment above r.e this. Also r.e the ones that are supported, you are incorrect on this, refer to https://www.zabbix.com/life_cycle_and_release_policy which shows versions as far back as Zabbix 3.0 LTS are still being supported under their Limited Support cycle for critical and security relevant updates.

Reply:
Ok also this new regex will have to be adjusted.....the way this is programmed at the moment, we could not log in successfully and because the resulting page will contain Zabbix in its response, and will return a 200 OK code, the code will think that the credentials were successful.

Reply:
Alright so I figured out that on Zabbix 5.x and later the `/profile.php` page got moved to `/zabbix.php?action=userprofile.edit`, so all one needs to do is slightly adjust the original regex and then ensure that if the original request to `/profile.php` fails, that the same test is tried again with `/zabbix.php?action=userprofile.edit`. If both attempts fail then we will return the usual failure case.